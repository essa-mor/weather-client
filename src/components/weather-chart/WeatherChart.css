.bottom_label{
	fill: #bababa;
	font-family: arial, sans-serif;
	font-size: 12px;
}

.bold{
	font-weight: bold;
}