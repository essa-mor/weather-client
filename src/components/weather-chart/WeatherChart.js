import React from 'react';

function WeatherChart({dt}) {
	return <div>Chart {dt}</div>;
}

export default WeatherChart;