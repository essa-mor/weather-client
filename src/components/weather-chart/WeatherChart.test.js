import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import WeatherChart from './WeatherChart';

const data = [
	{ dt: 1535889600, dt_txt: '2018-09-02 12:00:00', temp: 29.61 },
	{ dt: 1535900400, dt_txt: '2018-09-02 15:00:00', temp: 29.11 },
	{ dt: 1535911200, dt_txt: '2018-09-02 18:00:00', temp: 28.46 },
	{ dt: 1535922000, dt_txt: '2018-09-02 21:00:00', temp: 27.5 },
	{ dt: 1535932800, dt_txt: '2018-09-03 00:00:00', temp: 27.24 },
	{ dt: 1535943600, dt_txt: '2018-09-03 03:00:00', temp: 26.97 },
	{ dt: 1535954400, dt_txt: '2018-09-03 06:00:00', temp: 26.69 },
	{ dt: 1535965200, dt_txt: '2018-09-03 09:00:00', temp: 27.19 }];


describe('WeatherChart test', () => {
	it('Renders properly', () => {
		const tree = renderer.create(<WeatherChart data={data} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});