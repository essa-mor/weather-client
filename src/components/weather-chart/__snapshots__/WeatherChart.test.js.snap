// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherChart test Renders properly 1`] = `
<svg
  height={90}
  width={440}
>
  <text
    className="bottom_label bold"
    dy=".71em"
    x={10}
    y={5.179372197309419}
  >
    29.6
  </text>
  <text
    className="bottom_label"
    dy=".71em"
    x={65.71428571428571}
    y={8.542600896860996}
  >
    29.1
  </text>
  <text
    className="bottom_label"
    dy=".71em"
    x={121.42857142857142}
    y={12.914798206278022}
  >
    28.5
  </text>
  <text
    className="bottom_label"
    dy=".71em"
    x={177.14285714285714}
    y={19.372197309417047}
  >
    27.5
  </text>
  <text
    className="bottom_label"
    dy=".71em"
    x={232.85714285714283}
    y={21.12107623318387}
  >
    27.2
  </text>
  <text
    className="bottom_label"
    dy=".71em"
    x={288.57142857142856}
    y={22.93721973094172}
  >
    27.0
  </text>
  <text
    className="bottom_label"
    dy=".71em"
    x={344.2857142857143}
    y={24.82062780269058}
  >
    26.7
  </text>
  <text
    className="bottom_label"
    dy=".71em"
    x={400}
    y={21.45739910313901}
  >
    27.2
  </text>
  <path
    d="M0,20.17937219730942C20.95238095238095,21.289911623299908,41.9047619047619,22.400451049290393,62.857142857142854,23.542600896860996C83.80952380952381,24.6847507444316,104.76190476190473,25.858511013582312,125.71428571428571,27.91479820627802C146.66666666666669,29.97108539897373,167.6190476190476,32.90989951521445,188.57142857142856,34.37219730941705C209.52380952380952,35.834495103619645,230.47619047619048,35.82027657578415,251.42857142857142,36.12107623318387C272.38095238095235,36.42187589058359,293.33333333333337,37.03769373321854,314.2857142857143,37.93721973094172C335.2380952380952,38.8367457286649,356.1904761904761,40.01997988147633,377.1428571428571,39.82062780269058C398.09523809523813,39.62127572390483,419.04761904761904,38.03933741352192,440,36.45739910313901"
    fill="none"
    id="line"
    stroke="rgb(255,209,24)"
    strokeWidth="2"
  />
  <path
    d="M0,20.17937219730942C20.95238095238095,21.289911623299908,41.9047619047619,22.400451049290393,62.857142857142854,23.542600896860996C83.80952380952381,24.6847507444316,104.76190476190473,25.858511013582312,125.71428571428571,27.91479820627802C146.66666666666669,29.97108539897373,167.6190476190476,32.90989951521445,188.57142857142856,34.37219730941705C209.52380952380952,35.834495103619645,230.47619047619048,35.82027657578415,251.42857142857142,36.12107623318387C272.38095238095235,36.42187589058359,293.33333333333337,37.03769373321854,314.2857142857143,37.93721973094172C335.2380952380952,38.8367457286649,356.1904761904761,40.01997988147633,377.1428571428571,39.82062780269058C398.09523809523813,39.62127572390483,419.04761904761904,38.03933741352192,440,36.45739910313901L440,60C419.04761904761904,60,398.0952380952381,60,377.1428571428571,60C356.19047619047615,60,335.2380952380952,59.99999999999999,314.2857142857143,60C293.33333333333337,60.00000000000001,272.3809523809524,60.00000000000001,251.42857142857142,60C230.47619047619045,59.99999999999999,209.5238095238095,59.99999999999999,188.57142857142856,60C167.61904761904762,60.00000000000001,146.66666666666666,60,125.71428571428571,60C104.76190476190476,60,83.80952380952381,59.99999999999999,62.857142857142854,60C41.9047619047619,60.00000000000001,20.95238095238095,60,0,60Z"
    fill="rgba(255, 204, 0, 0.2)"
    id="area"
  />
  <text
    className="bottom_label"
    dy=".5em"
    x={10}
    y={75}
  >
    12:00
  </text>
  <text
    className="bottom_label"
    dy=".5em"
    x={65.71428571428571}
    y={75}
  >
    15:00
  </text>
  <text
    className="bottom_label"
    dy=".5em"
    x={121.42857142857142}
    y={75}
  >
    18:00
  </text>
  <text
    className="bottom_label"
    dy=".5em"
    x={177.14285714285714}
    y={75}
  >
    21:00
  </text>
  <text
    className="bottom_label"
    dy=".5em"
    x={232.85714285714283}
    y={75}
  >
    00:00
  </text>
  <text
    className="bottom_label"
    dy=".5em"
    x={288.57142857142856}
    y={75}
  >
    03:00
  </text>
  <text
    className="bottom_label"
    dy=".5em"
    x={344.2857142857143}
    y={75}
  >
    06:00
  </text>
  <text
    className="bottom_label"
    dy=".5em"
    x={400}
    y={75}
  >
    09:00
  </text>
</svg>
`;
